wyvern/components/models/model_component.py

```diff
@@ -150,6 +154,15 @@ def events_generator() -> List[ModelEvent]:
 
         return model_output
 
+    async def inference(
+        self,
+        input: MODEL_INPUT,
```

Comment:
this MODEL_INPUT will be a base model input class.

There will be a `MultiEntityModelInput` and `SingleEntityModelInput`.